lib/util/package-rules/merge-confidence.ts

```diff
@@ -10,6 +12,18 @@ export class MergeConfidenceMatcher extends Matcher {
     if (is.nullOrUndefined(matchConfidence)) {
       return null;
     }
+
+    /*
+     * Throw an error for unauthenticated use of the matchConfidence matcher.
+     */
+    if (is.undefined(getApiToken())) {
+      const error = new Error(MISSING_API_CREDENTIALS);
+      error.validationError = 'Missing credentials';
+      error.validationMessage =
+        "The 'matchConfidence' matcher in 'packageRules' requires authentication. Please refer to the documentation and add the required host rule.";
```

Comment:
can we add a link to the docs?

Reply:
We can do it "indirectly", but the result is not pretty.

For example
_____
There are missing credentials for the authentication-required feature. As a precaution, Renovate will pause PRs until it is resolved.

Error type: Missing credentials
Message: `The 'matchConfidence' matcher in 'packageRules' requires authentication. Please refer to the `[`documentation`](https://docs.renovatebot.com/configuration-options/#matchconfidence)` and add the required host rule.`
____

source markdown:
```markdown
 Please refer to the `[`documentation`](https://docs.renovatebot.com/configuration-options/#matchconfidence)` and add the required host rule.`
```

Reply:
how about this style instead or something similar ?

____
There are missing credentials for the authentication-required feature. As a precaution, Renovate will pause PRs until it is resolved.

Error type: The `matchConfidence` matcher in `packageRules` requires authentication. Please refer to the [documentation](https://docs.renovatebot.com/configuration-options/#matchconfidence) and add the required host rule.
Message:  `Missing credentials`

___

Reply:
@rarkins WDYT?

Reply:
[Current issue iteration](https://github.com/ladzaretti/grouping-mc-testing/issues/49)